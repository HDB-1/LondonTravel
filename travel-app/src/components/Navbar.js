import React from 'react';
import '../styling/Navbar.css';
import 'bootstrap/dist/css/bootstrap.min.css';
import { Navbar, Nav } from 'react-bootstrap';

function NavBar() {
    return (
        <div>
            <Navbar bg="primary" expand="lg" variant="dark">
                <Navbar.Brand href="">London Travel</Navbar.Brand>
                <Navbar.Toggle aria-controls="basic-navbar-nav" />
                <Navbar.Collapse id="basic-navbar-nav">
                    <Nav className="mr-auto">
                        <Nav.Link className="link" href="/" id="home_link">Home</Nav.Link>
                        <Nav.Link className="link" href="/linestatus" id="line_link">Lines</Nav.Link>
                        <Nav.Link className="link" href="/map" id="map_link">Tube Map</Nav.Link>
                        <Nav.Link className="link" href="/weather" id="weather_link">Weather</Nav.Link>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        </div>
    )
}

export default NavBar;