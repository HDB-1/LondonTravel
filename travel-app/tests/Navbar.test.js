import React from 'react';
import { shallow } from 'enzyme';
import { Link } from "react-router-dom";
import Navbar from '../src/components/Navbar';

describe("shallow Navbar", () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Navbar />)
    })
    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })
    it("Should render 4 <Link /> element", () => {
        expect(wrapper.find(".link").length).toEqual(4);
    })
    it("Should create a link to the '/' route.", () => {
        expect(wrapper.find("#home_link").props()).toHaveProperty("href", "/");
    })
    it("Should create a link to the '/linestatus' route.", () => {
        expect(wrapper.find("#line_link").props()).toHaveProperty("href", "/linestatus");
    })
    it("Should create a link to the '/map' route.", () => {
        expect(wrapper.find("#map_link").props()).toHaveProperty("href", "/map");
    })
    it("Should create a link to the '/weather' route.", () => {
        expect(wrapper.find("#weather_link").props()).toHaveProperty("href", "/weather");
    })
})