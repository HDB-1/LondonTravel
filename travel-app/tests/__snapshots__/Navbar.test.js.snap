// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow Navbar should render correctly 1`] = `
<div>
  <Navbar
    bg="primary"
    collapseOnSelect={false}
    expand="lg"
    variant="dark"
  >
    <NavbarBrand
      href=""
    >
      London Travel
    </NavbarBrand>
    <NavbarToggle
      aria-controls="basic-navbar-nav"
      label="Toggle navigation"
    />
    <NavbarCollapse
      id="basic-navbar-nav"
    >
      <Nav
        className="mr-auto"
        fill={false}
        justify={false}
      >
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
          href="/"
          id="home_link"
        >
          Home
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
          href="/linestatus"
          id="line_link"
        >
          Lines
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
          href="/map"
          id="map_link"
        >
          Tube Map
        </NavLink>
        <NavLink
          as={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "displayName": "SafeAnchor",
              "render": [Function],
            }
          }
          className="link"
          disabled={false}
          href="/weather"
          id="weather_link"
        >
          Weather
        </NavLink>
      </Nav>
    </NavbarCollapse>
  </Navbar>
</div>
`;
